File: src/main/java/net/lyragames/practice/utils/item/ItemListener.java
package net.lyragames.practice.utils.item;

import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerInteractEvent;

public class ItemListener implements Listener {

    @EventHandler
    public void onInteract(PlayerInteractEvent event) {
        CustomItemStack customItemStack = CustomItemStack.getByInteraction(event);

        if (customItemStack == null) return;

        if (customItemStack.isRightClick() && !event.getAction().name().contains("RIGHT")) {
            return;
        }

        customItemStack.getClicked().accept(event);

        if (customItemStack.isRemoveOnClick()) {
            CustomItemStack.getCustomItemStacks().remove(customItemStack);
        }
    }
}

File: src/main/kotlin/net/lyragames/practice/utils/item/ItemListener.kt
package net.lyragames.practice.utils.item

import net.lyragames.practice.utils.item.CustomItemStack.Companion.getByInteraction
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerInteractEvent

class ItemListener : Listener {
    @EventHandler
    fun onInteract(event: PlayerInteractEvent) {
        val customItemStack = getByInteraction(event) ?: return
        if (customItemStack.rightClick && !event.action.name.contains("RIGHT")) {
            return
        }
        customItemStack.clicked!!.accept(event)
        if (customItemStack.removeOnClick) {
            CustomItemStack.customItemStacks.remove(customItemStack)
        }
    }
}